samples/tutorials/mnist/04_Reusable_and_Pre-build_Components_as_Pipeline.ipynb

```diff
@@ -0,0 +1,799 @@
+    "GCS_BUCKET='gs://<ADD STORAGE LOCATION HERE>'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "tags": [
+     "parameter"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# Optional Parameters, but required for running outside kubeflow cluster\n",
+    "HOST = '<ADD HOST NAME TO TALK TO KUBEFLOW PIPELINE HERE>'\n",
+    "CLIENT_ID = '<ADD OAuth CLIENT ID USED BY IAP HERE>'\n",
+    "OTHER_CLIENT_ID = '<ADD OAuth CLIENT ID USED TO OBTAIN AUTH CODES HERE>'\n",
+    "OTHER_CLIENT_SECRET = '<ADD OAuth CLIENT SECRET USED TO OBTAIN AUTH CODES HERE>'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create client\n",
+    "\n",
+    "**If submit outside the kubeflow cluster, need the following**\n",
+    "- `host`: the host name to use to talk to Kubeflow Pipelines, i.e., \"https://`<your-deployment>`.endpoints.`<your-project>`.cloud.goog/pipeline\"\n",
+    "- `client_id`: The client ID used by Identity-Aware Proxy\n",
+    "- `other_client_id`: The client ID used to obtain the auth codes and refresh tokens.\n",
+    "- `other_client_secret`: The client secret used to obtain the auth codes and refresh tokens.\n",
+    "- For getting `other_client_id` and `other_client_secret`, you'll need to create OAuth client ID credentials of type `Other` according to the tutorial [here](\n",
+    "https://cloud.google.com/iap/docs/authentication-howto#authenticating_from_a_desktop_app)\n",
+    "\n",
+    "```python\n",
+    "client = kfp.Client(host, client_id, other_client_id, other_client_secret)\n",
+    "```\n",
+    "\n",
+    "**If you run and submit within the kubeflow cluster**, the following is enough\n",
+    "```python\n",
+    "client = kfp.Client()\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create kfp client\n",
+    "in_cluster = True\n",
+    "try:\n",
+    "  k8s.config.load_incluster_config()\n",
+    "except:\n",
+    "  in_cluster = False\n",
+    "  pass\n",
+    "\n",
+    "if in_cluster:\n",
+    "    client = kfp.Client()\n",
+    "else:\n",
+    "    client = kfp.Client(host=HOST, \n",
+    "                        client_id=CLIENT_ID,\n",
+    "                        other_client_id=OTHER_CLIENT_ID, \n",
+    "                        other_client_secret=OTHER_CLIENT_SECRET)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Build reusable components"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Writing the program code"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following cell creates a file `app.py` that contains a Python script. The script takes a GCS bucket name as an input argument, gets the lists of blobs in that bucket, prints the list of blobs and also writes them to an output file."
```

Comment:
GCS -> Cloud Storage
list of blobs and also writes -> list of blobs, and writes

Reply:
This has been changed to align with previous notebooks.